website/docs/core-concepts/components/remote-state.md

```diff
@@ -0,0 +1,337 @@
+
+components:
+  terraform:
+    vpc-flow-logs-bucket-1:
+      metadata:
+        # Point to the Terraform component in `components/terraform` folder
+        component: infra/vpc-flow-logs-bucket
+        inherits:
+          # Inherit all settings and variables from the 
+          # `vpc-flow-logs-bucket/defaults` base Atmos component
+          - vpc-flow-logs-bucket/defaults
+      vars:
+        # Define variables that are specific for this component
+        # and are not set in the base component
+        name: vpc-flow-logs-bucket-1
+        # Override the default variables from the base component
+        traffic_type: "REJECT"
+```
+
+<br/>
+
+Having the stacks configured as shown above, we can now provision the `vpc-flow-logs-bucket-1` Atmos component into the `ue2-dev` stack by executing
+the following Atmos commands:
+
+```shell
+atmos terraform plan vpc-flow-logs-bucket-1 -s ue2-dev
+atmos terraform apply vpc-flow-logs-bucket-1 -s ue2-dev
+```
+
+<br/>
+
+## Configure and Provision `vpc` Component
+
+Having the `vpc-flow-logs-bucket` Terraform component provisioned into the `ue2-dev` stack, we can now configure the `vpc` Terraform component
+to obtain the outputs from the remote state of the `vpc-flow-logs-bucket-1` Atmos component.
+
+In the `components/terraform/infra/vpc/remote-state.tf` file, configure the
+[remote-state](https://github.com/cloudposse/terraform-yaml-stack-config/tree/main/modules/remote-state) Terraform module to obtain the remote state
+for the `vpc-flow-logs-bucket-1` Atmos component:
+
+```hcl title="components/terraform/infra/vpc/remote-state.tf"
+module "vpc_flow_logs_bucket" {
+  count = var.vpc_flow_logs_enabled ? 1 : 0
+
+  source  = "cloudposse/stack-config/yaml//modules/remote-state"
+  version = "1.3.1"
+
+  # Specify the Atmos component name (defined in YAML stack config files) 
+  # for which to get the remote state outputs
+  component = var.vpc_flow_logs_bucket_component_name
+
+  # Override the context variables to point to a different Atmos stack if the 
+  # `vpc-flow-logs-bucket-1` Atmos component is provisioned in another AWS account, OU or region
+  stage       = try(coalesce(var.vpc_flow_logs_bucket_stage_name, module.this.stage), null)
+  tenant      = try(coalesce(var.vpc_flow_logs_bucket_tenant_name, module.this.tenant), null)
+  environment = try(coalesce(var.vpc_flow_logs_bucket_environment_name, module.this.environment), null)
+
+  # `context` input is a way to provide the information about the stack (using the context
+  # variables `namespace`, `tenant`, `environment`, `stage` defined in the stack config)
+  context = module.this.context
+}
+```
+
+In the `components/terraform/infra/vpc/vpc-flow-logs.tf` file, configure the `aws_flow_log` resource for the `vpc` Terraform component to use the
+remote state output `vpc_flow_logs_bucket_arn` from the `vpc-flow-logs-bucket-1` Atmos component:
+
+```hcl title="components/terraform/infra/vpc/remote-state.tf"
```

Comment:
I'm concerned about the maintainability of this example. Does docusaurus have a way to import a file into a code block?

Reply:
ideally, we would have a folder with a working example, and then import the files here surrounded by decriptions.